File: java/pattern/src/templatemethod/BankTemplate.java
package templatemethod;

public abstract class BankTemplate {
    // 具体方法
    public void takeNumber() {
        System.out.println("取号排队");
    }

    // 具体办理的业务 钩子方法
    public abstract void transact();

    public void evalution() {
        System.out.println("反馈评分");
    }

    /**
     * 模板方法
     */
    public final void process() {
        this.takeNumber();
        this.transact();
        this.evalution();
    }
}

File: php/templatemethod/BankTemplate.php
<?php

/**
 * 去银行办理业务模板
 */
abstract class BankTemplate
{
    /**
     * 具体方法
     * @return [type] [description]
     */
    public function takeNumber()
    {
        echo '取号排队', PHP_EOL;
    }

    /**
     * 具体办理的业务 钩子方法
     * @return [type] [description]
     */
    abstract public function transact();

    public function evalution()
    {

        echo '反馈评分', PHP_EOL;
    }

    /**
     * 模板方法
     */
    public function process()
    {
        $this->takeNumber();
        $this->transact();
        $this->evalution();
    }
}
